nomos-services/network/src/backends/libp2p/command.rs

```diff
@@ -25,7 +25,7 @@ pub enum Command {
     StreamSend {
         peer_id: PeerId,
         protocol: StreamProtocol,
-        packet_body: PacketBody,
+        data: Box<dyn AsyncWritable + Send + Sync>,
```

Comment:
So, I think in this case what you really want data is to be something you can asynchronously be read into bytes (https://docs.rs/futures/latest/futures/io/trait.AsyncRead.html ?). I do not think the data itself need to know how to write itself into whatever. That should be solved in the receiver side of this message. This would completely decouple the serialization/reading from the writing itself. I do not know if I explained it well here. Or maybe there are more reasons for the `AsyncWritable` trait?

Reply:
I agree with your point. I also felt that it’s not natural to define `write_to` in the data side. Actually we had done something similar in the previous mixnet implementation last year, but I agree that it’s more natural to implement AsyncRead or something similar. 

The initial idea why I introduced AsyncWritable is that I didn’t want to serialize PacketBody into a single Vec first and write the Vec to the writer (stream). 
```
flag | len | body
```
Instead, I thought it’s better to write these pieces to the writer directly.

Reply:
Let me try the way you suggested. Reading bytes from PacketBody and piping them to the writer directly.

Reply:
The vec is not a bad solution. Is `move` which solves a lot of problems. It also clears all the async here which probably is not necessary unless you are really reading the packet asyncrhonously at that stage. Otherwise the async code here is just sync 😅

Reply:
Yes. Only async part is libp2p::Stream that implements futures::AsyncWrite. Except for it, all other things are sync. 

>  The vec is not a bad solution. Is move which solves a lot of problems.

Does this mean that we can serialize a PacketBody into a Vec by consuming the PacketBody, so that byte pieces in the PacketBody are moved (not copied) into the Vec?